src/main/java/io/cryostat/net/web/http/api/v1/RecordingsPostHandler.java

```diff
@@ -174,12 +182,17 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
                         ? 0
                         : Integer.parseInt(m.group(4).substring(1));
 
+        ConnectionDescriptor connectionDescriptor = getConnectionDescriptorFromContext(ctx);
```

Comment:
How does this work? `getConnectionDescriptorFromContext` relies on retrieving the `targetId` path parameter, but this API handler doesn't have such a path parameter. I think you'll end up with a `ConnectionDescriptor` with a null targetId and null credentials, no?

Reply:
So the `ConnectionDescriptor` fields are populated using the path parameters, which is why this approach works for `TargetRecordingPatchSave` but not here, right? 

In that case, how would I organize the directory structure since here the recording is being uploaded directly from the client? I believe the only target identifier I have is `TARGET_NAME` in the file format `[$TARGET_NAME]_[$RECORDING_NAME]_[$DATE]T[$TIME]Z[.$COUNTER][.jfr]`.

Reply:
To add on, my changes to the other handlers are all centered around recordings being organized in base 32 encoded `ServiceUri` (i.e. `targetId`) directories.

Reply:
Right, it works elsewhere in handlers that do have a `targetId` path parameter, but it won't work here unfortunately.

Can you add a special-cased subdirectory for these client re-uploaded recordings? The actual subdirectories would always have names that are base32-encoded ServiceURIs (or maybe base32-encoded host:port pairs), so simply naming the directory an unencoded `"unlabelled"` or something could work, I think. You can upload into that and also merge those into the results when scanning the filesystem for all archived recordings.

At some point in a separate PR we should add a V2 handler that allows for re-uploads of recordings that does have a targetId path parameter so that the uploads can be associated with a particular target. Then we can also make it possible to query for archived recordings by target, and then the web-client can display archived recordings per-target as well.

Reply:
That's a good idea and it's a simple add-on; I'll go ahead and do that. 

As for the V2 handler, I can get started on that if you'd like once I finish this PR and have helped Janelle with the API IT coverage.